src/Aspire.Hosting/ResourceBuilderExtensions.cs

```diff
@@ -363,6 +363,7 @@ private static void ApplyEndpoints<T>(this IResourceBuilder<T> builder, IResourc
     /// <param name="callback">Callback that modifies the endpoint.</param>
     /// <param name="createIfNotExists">Create endpoint if it does not exist.</param>
     /// <returns></returns>
+    [System.Diagnostics.CodeAnalysis.SuppressMessage("ApiDesign", "RS0026:Do not add multiple public overloads with optional parameters", Justification = "<Pending>")]
     public static IResourceBuilder<T> WithEndpoint<T>(this IResourceBuilder<T> builder, string endpointName, Action<EndpointAnnotation> callback, bool createIfNotExists = true) where T : IResourceWithEndpoints
```

Comment:
Maybe we turn this into three methods:

```csharp
public static IResourceBuilder<T> WithEndpoint<T>(this IResourceBuilder<T> builder, string endpointName, Action<EndpointAnnotation> callback) where T : IResourceWithEndpoints {}
public static IResourceBuilder<T> WithExistingEndpoint<T>(this IResourceBuilder<T> builder, string endpointName, Action<EndpointAnnotation> callback) where T : IResourceWithEndpoints {}
public static IResourceBuilder<T> WithEndpoint<T>(this IResourceBuilder<T> builder, int? port = null, int? targetPort = null, string? scheme = null, string? name = null, string? env = null, bool isProxied = true, bool? isExternal = null) where T : IResource {}
```

Basically if you want to use the callback against an existing endpoint you would use `WithExistingEndpoint(...)`.